src/lib/receiver.js

```diff
@@ -186,16 +203,31 @@ function createReceiver (opts) {
     }
 
     // The payment request is extracted from the ilp_header
-    let packet = transfer.data && transfer.data.ilp_header
+    const packet = transfer.data && transfer.data.ilp_header
 
     if (!packet) {
       debug('got notification of transfer with no packet attached')
       return rejectIncomingTransfer(transfer.id, 'no-packet')
     }
 
-    // check if the address contains "~ipr" + our receiver id
-    if (packet.account.indexOf(receiverId) === -1) {
-      debug('got notification of transfer for another receiver: ' + packet.account + ' my id is: ' + receiverId)
+    // check if the address starts with our address
+    if (packet.account.indexOf(getAddress()) !== 0) {
```

Comment:
Hmm, interesting. Right now it seems that there is 1:1:1 relationship between receiver, client and plugin. (See `getAddress()`.)

If we switch to a multi-plugin receiver, we would still know which plugin emitted the event, so we'd still know what our address is supposed to be. So we could check it. I can't think of a strong argument for or against verifying the address. Maybe someone else has one?